features/FEATURE_BLE/ble/Gap.h

```diff
@@ -321,6 +269,617 @@ class GapAdvertisingData;
+     *
+     * @param handle Advertising set handle.
+     * @param periodicAdvertisingIntervalMin Minimum interval for periodic advertising.
+     * @param periodicAdvertisingIntervalMax Maximum interval for periodic advertising.
+     * @param advertiseTxPower Include transmission power in the advertisements.
+     * @return BLE_ERROR_NONE on success.
+     */
+    virtual ble_error_t setPeriodicAdvertisingParameters(
+        ble::advertising_handle_t handle,
+        ble::periodic_interval_t periodicAdvertisingIntervalMin,
+        ble::periodic_interval_t periodicAdvertisingIntervalMax,
+        bool advertiseTxPower = true
+    );
+
+    /** Set new periodic advertising payload for a given advertising set.
+     *
+     * @param handle Advertising set handle.
+     * @param payload Advertising payload.
+     * @return BLE_ERROR_NONE on success.
+     */
+    virtual ble_error_t setPeriodicAdvertisingPayload(
+        ble::advertising_handle_t handle,
+        mbed::Span<uint8_t> payload
+    );
+
+    /** Start periodic advertising for a given set. Periodic advertising will not start until
+     *  normal advertising is running but will continue to run after normal advertising has stopped.
+     *
+     * @param handle Advertising set handle.
+     * @return BLE_ERROR_NONE on success.
+     */
+    virtual ble_error_t startPeriodicAdvertising(ble::advertising_handle_t handle);
+
+    /** Stop periodic advertising for a given set.
+     *
+     * @param handle Advertising set handle.
+     * @return BLE_ERROR_NONE on success.
+     */
+    virtual ble_error_t stopPeriodicAdvertising(ble::advertising_handle_t handle);
+
+    /** Check if periodic advertising is active for a given advertising set.
+     *
+     * @param handle Advertising set handle.
+     * @return True if periodic advertising is active on this set.
+     */
+    virtual bool isPeriodicAdvertisingActive(ble::advertising_handle_t handle);
+
+    /*                                     scanning                                              */
+
+    /** Set new scan parameters.
+     *
+     * @param params Scan parameters, @see GapScanParameters for details.
+     * @return BLE_ERROR_NONE on success.
+     */
+    virtual ble_error_t setScanParameters(const ble::ScanParameters& params);
+
+    /** Start scanning.
+     *
+     * @param filtering Filtering policy.
+     * @param duration How long to scan for. Special value 0 means scan forever.
+     * @param period How long to scan for in single period. If the period is 0 and duration
+     *               is nonzero the scan will last for single duration.
+     *
+     * @note When the duration and period parameters are non-zero scanning will last for
+     * the duration within the period. After the scan period has expired a new scan period
+     * will begin and scanning. This will repeat until stopScan() is called.
+     *
+     * @return BLE_ERROR_NONE on success.
+     */
+    virtual ble_error_t startScan(
```

Comment:
I'm not sure how easy that would be to add to the `Duration` class, but could we have a special `Duration::forever` value?

Reply:
That is something we can add when it is meaningful: 

```c++
scan_duration_t::forever(); // works 
scan_interval_t::forever(); // compile time error as there is no special forever value for scan interval.
```

@donatieng What do you think ?

Reply:
Sounds good :)